phys/module_bl_ysu.F

```diff
@@ -206,10 +241,30 @@ subroutine ysu(u3d,v3d,th3d,t3d,qv3d,qc3d,qi3d,p3d,p3di,pi3d,               &
                                                                         dvsfc, &
                                                                         dtsfc, &
                                                                         dqsfc
-
+   real,dimension(ims:ime,kms:kme,jms:jme) :: a_u,a_v,a_t,a_e,b_u,b_v,b_t,b_e, &
+                                                     a_q,b_q,dlg,dl_u,sfk,vlk
+   real,dimension(ims:ime,jms:jme)         :: frcurb
+   real :: bepswitch ! 0 if not using bep or bep+bem, 1 if using
 !
    qv2d(its:ite,:) = 0.0
 !
+   bepswitch = 0.0
+   a_u(:,:,:)=0.0
+   a_v(:,:,:)=0.0
+   a_t(:,:,:)=0.0
+   a_q(:,:,:)=0.0
+   a_e(:,:,:)=0.0
+   b_u(:,:,:)=0.0
+   b_v(:,:,:)=0.0
+   b_t(:,:,:)=0.0
+   b_q(:,:,:)=0.0
+   b_e(:,:,:)=0.0
+   sfk(:,:,:)=1.0
+   vlk(:,:,:)=1.0
+   dl_u(:,:,:)=0.0
+   dlg(:,:,:)=0.0
+   frcurb(:,:)=0.0   
+
```

Comment:
Are these initializations necessary? 20 lines down, if the "flag_bep" is true, then they are all initialized.

Reply:
Yes, they are necessary for the initialization when "flag_bep" is false. I wanted to have a default initialization for the way most users will run (without BEP).